telethon_generator/tl_generator.py

```diff
@@ -687,21 +687,17 @@ def write_request_result_code(builder, tlobject):
                          will be written
         """
         if tlobject.result.startswith('Vector<'):
-            # Vector results are a bit special since they can also be composed
-            # of integer values and such; however, the result of requests is
-            # not parsed as arguments are and it's a bit harder to tell which
-            # is which.
-            if tlobject.result == 'Vector<int>':
+            # Check if Vector contents are boxed or not
+            # We're implying that constructor names always
+            # start with lowecase and types start with uppercase
+            match = re.match(r'Vector<(.*)>', tlobject.result)
+            if match and match.group(1).lower:
+                inner_type = match.group(1)
+                # not boxed
                 builder.writeln('reader.read_int()  # Vector id')
                 builder.writeln('count = reader.read_int()')
                 builder.writeln(
-                    'self.result = [reader.read_int() for _ in range(count)]'
-                )
-            elif tlobject.result == 'Vector<long>':
-                builder.writeln('reader.read_int()  # Vector id')
-                builder.writeln('count = reader.read_long()')
-                builder.writeln(
-                    'self.result = [reader.read_long() for _ in range(count)]'
+                    'self.result = [reader.read_{}() for _ in range(count)]'.format(inner_type)
```

Comment:
`.read_string()` won't be a thing :)
https://github.com/LonamiWebs/Telethon/blob/0e0bc6ecbc84e2703aa75bc7e626ff13a2bf4032/telethon/extensions/binary_reader.py#L101

Reply:
Why not?..

Reply:
Because it's called `.tgread_string()`, not `.read_string()`.

Reply:
@Lonami why?..

Reply:
Because the format in which strings are read is related to Telegram itself, and not a "normal" binary mode, hence the `.tg` prefix.